File: workflow-payment/src/main/java/com/acme/workflows/payment/java/PaymentWorkflowImpl.java
package com.acme.workflows.payment.java;

import com.acme.contracts.services.notification.BookingRequest;
import com.acme.common.AbstractWorkflow;
import com.acme.contracts.workflows.invoicing.User;
import com.acme.contracts.workflows.payment.PaymentWorkflow;

public class PaymentWorkflowImpl extends AbstractWorkflow implements PaymentWorkflow {
    @Override
    public void getDeposit(BookingRequest request) {
        log("Request " + request.getId() + ": Running GetDeposit");
    }

    @Override
    public void getPayment(User user, double amount) {
        log("Get payment " + amount + " from User " + user.getId());
    }
}

File: workflow-payment/src/main/kotlin/com/acme/workflows/payment/kotlin/PaymentWorkflowImpl.kt
package com.acme.workflows.payment.kotlin

import com.acme.contracts.services.notification.BookingRequest
import com.acme.common.AbstractWorkflow
import com.acme.contracts.workflows.invoicing.User
import com.acme.contracts.workflows.payment.PaymentWorkflow

class PaymentWorkflowImpl : AbstractWorkflow(),
    PaymentWorkflow {
    override fun getDeposit(request: BookingRequest) {
        log("Request ${request.id}: Running ${::getDeposit.name}" )
    }

    override fun getPayment(user: User, amount: Double) {
        log("Get payment $amount from User ${user.id}" )
    }
}